.ci/travis/travis.sh

```diff
@@ -264,6 +264,50 @@ no-error-test-sbe)
   removeFolderWithProtectedFiles simple-binary-encoding
   ;;
 
+no-error-xwiki)
+  # non-snapshot version until https://github.com/checkstyle/checkstyle/issues/9236
+  # CS_POM_VERSION=$(mvn -e -q -Dexec.executable='echo' -Dexec.args='${project.version}' \
+  #                   --non-recursive org.codehaus.mojo:exec-maven-plugin:1.3.1:exec)
+  CS_POM_VERSION=8.39
+  echo version:$CS_POM_VERSION
+  mvn -e clean install -Pno-validations
+  mkdir -p .ci-temp/
+  cd .ci-temp/
+  git clone --depth 1 https://github.com/xwiki/xwiki-commons.git
```

Comment:
@romani Will `--depth 1` break local user run when the repo already exists? Why isn't this using `checkout_from`?

Reply:
because we never made it common for all CIs
```
07:43 $ ag --hidden "checkout_from"
.ci/wercker.sh
5:function checkout_from {
49:  checkout_from https://github.com/viesure/blog-sonar-build-breaker.git
64:  checkout_from https://github.com/pgjdbc/pgjdbc.git
78:  checkout_from https://github.com/Hipparchus-Math/hipparchus.git
85:  checkout_from https://github.com/CS-SI/Orekit.git
101:  checkout_from https://github.com/checkstyle/apex-core
112:  checkout_from https://github.com/jqno/equalsverifier.git
123:  checkout_from https://github.com/hibernate/hibernate-search.git
137:  checkout_from https://github.com/HtmlUnit/htmlunit
159:  checkout_from https://github.com/sevntu-checkstyle/sevntu.checkstyle.git
172:  checkout_from https://github.com/checkstyle/contribution.git
190:  checkout_from https://github.com/sevntu-checkstyle/methods-distance.git
203:  checkout_from https://github.com/OpenGamma/Strata.git
219:  checkout_from https://github.com/spring-projects/spring-integration.git
233:  checkout_from https://github.com/googlecloudplatform/spring-cloud-gcp
246:  checkout_from https://github.com/checkstyle/contribution.git
261:  checkout_from https://github.com/checkstyle/contribution.git
277:  checkout_from https://github.com/checkstyle/contribution.git
292:  checkout_from https://github.com/checkstyle/contribution.git
306:  checkout_from https://github.com/checkstyle/contribution.git
320:  checkout_from https://github.com/checkstyle/contribution.git
334:  checkout_from https://github.com/checkstyle/contribution.git
348:  checkout_from https://github.com/checkstyle/contribution.git
362:  checkout_from https://github.com/checkstyle/contribution.git
376:  checkout_from https://github.com/checkstyle/contribution.git
392:  checkout_from https://github.com/checkstyle/contribution.git
414:  checkout_from https://github.com/checkstyle/contribution.git
438:  checkout_from https://github.com/checkstyle/contribution.git

.ci/shippable.sh
5:function checkout_from {
39:  checkout_from https://github.com/checkstyle/contribution.git
55:  checkout_from https://github.com/checkstyle/contribution.git
75:  checkout_from https://github.com/checkstyle/contribution.git
92:  checkout_from https://github.com/checkstyle/contribution.git
110:  checkout_from https://github.com/checkstyle/contribution.git

```

, yes, re-execution on local gives error:
```
[INFO] Finished at: 2021-02-10T07:42:27-08:00
[INFO] ------------------------------------------------------------------------
fatal: destination path 'xwiki-commons' already exists and is not an empty directory.
✘-128 ~/java/github/romani/checkstyle [tmortagne/master L|✔] 
```

we need separate issue on this.

Reply:
> Why isn't this using checkout_from?

I don't see any reference to `checkout_from` in this file but I see no-error-test-sbe uses `git clone` for example. Since there is no point in downloading the whole history I tough it was better to download only the current version.

> when the repo already exists

Is it really possible ? Isn't each supposed to be executed on a clean slate ? Never used Travis myself that would sounds weird for a CI system like this to keep files form a previous run.

Reply:
https://github.com/checkstyle/checkstyle/issues/9268

Reply:
@tmortagne 
>  Isn't each new build supposed to be executed on a clean slate ? 

We are talking  about executing the command locally, not in CI. When CI fails or has issues, we tell users to run the command locally to produce and debug the same results.

Reply:
OK, it was not clear in your first message and I did not had this use case in mind. I can easily add workarounds to deal with that but won't `no-error-test-sbe` have the same issue for example (did not really look at other tasks) ?

Reply:
Yes, it was more for romani since he would know what I was talking about. It was to see if we wanted to fix it before merging, but he wanted it merged first.

`no-error-test-sbe` doesn't have the depth parameter set so it is not clear how it would have the same issue. You would need to elaborate.

Reply:
> no-error-test-sbe doesn't have the depth parameter set so it is not clear how it would have the same issue. You would need to elaborate.

I don't really see why `--depth 1` is a problem, when you use git clone on an already existing folder git will fail telling you that there is already a clone whatever parameter you use.

Reply:
@tmortagne You are correct, this is one of things the `checkout_from` function does. It checks if the folder exists and just does a fetch instead of cloning again.
https://github.com/checkstyle/checkstyle/blob/8f2db18d0ad3fd9883f38ce980955cca15be61b0/.ci/shippable.sh#L10-L17

I have limited experience with `depth` but I don't believe you can fetch on it for updates.

Reply:
Honestly since there is no point in downloading the whole history I would simply `rm -rf` before the clone in case there is already one.